Automattic/jetpack
# My Jetpack: Set post-checkout redirect URL on the plan purchase link

Fixes #27559.

#### Changes proposed in this Pull Request:

* Set a post-checkout redirect destination on the "Purchase a plan" link, so the user is redirected back to the "My Jetpack" screen after buying a plan

#### Other information:


#### Jetpack product discussion



#### Does this pull request change what data or activity we track or use?



No.

#### Testing instructions:






* Install and activate the Jetpack VideoPress standalone plugin
* Make sure you don't have the full Jetpack plugin active
* Make sure you are not logged into Jetpack Cloud
* Go to `Jetpack > VideoPress` and choose to start using a free plan on VideoPress

<img width="1151" alt="Screen Shot 2022-12-01 at 13 53 52" src="https://user-images.githubusercontent.com/6760046/205113023-160c5494-7a1b-486d-b8a7-7936be414a56.png">

* Go to `Jetpack > My Jetpack`
* On the bottom of the page, click the "Purchase a plan" link

<img width="300" alt="Screen Shot 2022-12-01 at 13 57 50" src="https://user-images.githubusercontent.com/6760046/205113424-acb68562-df19-4a5e-b3f9-f50d45cdf93c.png">

* You will land on the page offering products and bundles
* On the "Products" section, select the VideoPress plan and proceed with the checkout 

<img width="1016" alt="Screen Shot 2022-12-01 at 13 58 51" src="https://user-images.githubusercontent.com/6760046/205113852-4a8b8957-8ada-4c58-bd45-04d0565ab801.png">

* After you finish the checkout, you will be redirected to a page inside your site
* Before the change, the page would be `/wp-admin/admin.php?page=jetpack#/recommendations`, a page that would not exists on our test scenario since Jetpack is not installed or enabled
* After the change, the page will be the My Jetpack page, that was the page where the user started the purchase flow
